src/strategy/exponential_backoff.rs

```diff
@@ -1,14 +1,16 @@
 use std::time::Duration;
 use std::iter::Iterator;
-use std::u64::{MAX as U64_MAX};
+use std::u64::MAX as U64_MAX;
 
 /// A retry strategy driven by exponential back-off.
 ///
 /// The power corresponds to the number of past attempts.
-#[derive(Clone)]
+#[derive(Debug, Clone)]
 pub struct ExponentialBackoff {
     current: u64,
-    base: u64
+    base: u64,
+    factor: Option<u64>,
```

Comment:
If we have a sane default for the factor (`1`), why not just go with `u64` here instead of an `Option<u64>`, and set to 1 in `from_millis`?